### ISSUE TITLE
sql.mock broken behaviour in 0.8.1

### ISSUE THREAD

**Issue by darky** [xialvjun/ts-sql-plugin#27]
@xialvjun, in 0.7.1 `sql.mock` passed `placeholder` forwarded to sql "as is". Example:

```ts
sql`select * from users ${sql.mock({mock: '', placeholder: 'where user_id = 1'})}`
// generated sql in runtime - select * from users where user_id = 1
```
But on 0.8.1 `sql.mock` try to interpolate passed argument

```ts
sql`select * from users ${sql.mock<''>('where user_id = 1')}`
// generated sql in runtime - select * from users ??
```

Hot achieve old behaviour in 0.8.1?

**Comment by xialvjun** [comment]
``sql.mock<''>(sql.raw`where user_id = 1`))`` or `sql.mock<''>(sql.raw(['where user_id = 1']))`

Why I change it to this: old version is in fact `mock + raw`... But what if what we need is not raw string.

Imagine this case:
there is a database driver can escape a js object to complex sql value like `json->xxxx`...

We want to use `ts-sql-plugin/sql` to generate an object like: `{text: 'select * from person where name = ??', values: [{xxx: 'darky' }] }`, and let the database driver use the object to generate complete sql `select * from person where name = json->'xxx'->'darky'`...(ignore the sql inaccuracy)

We need to let the database driver do the escape process, if we transform conditions to string by ourselves, it's error-prone.
So we should not pass the runtime a string. We should pass the runtime the identity it.

**Comment by xialvjun** [comment]
Besides, new version can:

```ts
type Tables = 'person' | 'book' | 'store' | 'country';
let which_table: Table = 'person';
sql`select * from ${sql.mock<Tables>(sql.raw([which_table]))} where id = ${item_id}`;
```

**Comment by darky** [comment]
@xialvjun thanks! `sql.raw([which_table])` works!
But I make little type fixes for it, which will be included in release after `0.8.1`
